src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,57 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int CAPACITY = 10;
+
+    public static final class Entry<K, V> {
+        private final K key;
+        private V value;
+
+        public Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private final Entry<K, V>[] array = (Entry<K, V>[]) new Entry[CAPACITY];
+    private int size = 0;
```

Comment:
int type = 0 by default
```suggestion
    private int size;
```